### PR TITLE
[PICK] Create Elasticsearch client on the fly instead of caching it

### PR DESCRIPTION
**Pull request by Brian-McM**
Pick of https://github.com/tigera/operator/pull/1192

This commit changes logstorage to create the Elasticsearch client for every reconcile. This has been changed because we could be reconciling Elasticsearch credentials being recreated, in which case the cached client wouldn't work.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by lmm — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Pull request merged**
